src/robot_interface/robot_interface.py

```diff
@@ -11,129 +9,85 @@ class RobotInterface(metaclass=ABCMeta):
     """Interface to communicate with robots."""
 
     @abstractmethod
-    def schedule_task(self, task: Task) -> bool:
-        """Schedules a Task on the robot.
-
-        The method must adapt the standard mission to the
-        specific robots mission planning.
+    def initiate_task(self, task: Task) -> bool:
+        """Send a task to the robot and start the execution of the task
 
         Parameters
         ----------
         task : Task
-            Task object describing the mission task.
+            The task that should be initiated on the robot.
 
         Returns
         -------
         bool
-            True if successful of scheduling a task, false otherwise.
-        """
-        raise NotImplementedError
-
-    @abstractmethod
-    def mission_scheduled(self) -> bool:
-        """Determines if a mission is scheduled on the robot.
+            Indicating if the task was succesfully initiated
 
-        If a mission is scheduled it may be that another mission
-        should not be scheduled.
+        Raises
+        ------
+        RobotException
+            If the task is not initiated.
 
-        Returns
-        -------
-        bool
-            True if a mission is already scheduled, false otherwise.
         """
         raise NotImplementedError
 
     @abstractmethod
-    def task_status(self, task_id: Optional[UUID]) -> TaskStatus:
-        """Retrieves status of the task with the given id. If task id is not specified,
-        the status of the current executing mission task is returned.
+    def task_status(self) -> TaskStatus:
+        """Gets the status of the currently active task on robot.
 
         Parameters
         ----------
-        task_id : UUID
-            Unique identifier for the mission task which the status should be checked for.
+        None
 
         Returns
         -------
         TaskStatus
-            Enum member indicating the current mission status.
-        """
-        raise NotImplementedError
-
-    @abstractmethod
-    def abort_mission(self) -> bool:
-        """Aborts the current mission for the robot.
+            Status of the execution of current task.
 
-        If another mission is queued, it should not start before a verification
-        is sent.
+        Raises:
+        ------
+        RobotException
+            If the task status can't be retrived.
 
-        Returns
-        -------
-        bool
-            True on successful abort of mission, false otherwise.
         """
         raise NotImplementedError
 
     @abstractmethod
-    def log_status(self, task_status: TaskStatus, current_task: Task) -> None:
-        """Logs the status of the executing task that is being monitored.
+    def stop(self) -> bool:
```

Comment:
Description of what this method returns and what it actually returns does not match. Think we should return `None` and throw exception on failing